From de917894fecdc6606c68ddf352be72de07185a68 Mon Sep 17 00:00:00 2001
From: yanhaodong <haodong.yan@spacemit.com>
Date: Thu, 19 Dec 2024 17:58:43 +0800
Subject: [PATCH 1371/1448] Revert "i2c: spacemit/k1: update i2c function while
 reboot"

we dont have to send stop between the two msgs of readmsg.
so we use the method while reboot before.

This reverts commit 11f6a2bc9736a02fb055d78ae2cbb3dfc0090c63.

Change-Id: I8510c0e936dd34ba2dd0c30f9e40b15eb909ebe7
---
 drivers/i2c/busses/i2c-k1x.c | 27 ++++++---------------------
 1 file changed, 6 insertions(+), 21 deletions(-)

diff --git a/drivers/i2c/busses/i2c-k1x.c b/drivers/i2c/busses/i2c-k1x.c
index 363e150eabbb..f1761704f182 100644
--- a/drivers/i2c/busses/i2c-k1x.c
+++ b/drivers/i2c/busses/i2c-k1x.c
@@ -1120,9 +1120,8 @@ static void spacemit_i2c_init_xfer_params(struct spacemit_i2c_dev *spacemit_i2c)
 static int spacemit_i2c_pio_xfer(struct spacemit_i2c_dev *spacemit_i2c)
 {
 	int ret = 0, xfer_try = 0;
-	u32 status, ctrl;
+	u32 status;
 	signed long timeout;
-	signed long stop_timeout = 3000;
 
 xfer_retry:
 	/* calculate timeout */
@@ -1153,17 +1152,7 @@ static int spacemit_i2c_pio_xfer(struct spacemit_i2c_dev *spacemit_i2c)
 	}
 
 	while (spacemit_i2c->num > 0 && timeout > 0) {
-		ctrl = spacemit_i2c_read_reg(spacemit_i2c, REG_CR);
-		if (ctrl & CR_STOP)
-			while (stop_timeout > 0) {
-				status = spacemit_i2c_read_reg(spacemit_i2c, REG_SR);
-				udelay(100);
-				stop_timeout -= 100;
-				if (status & SR_MSD)
-					break;
-			}
-		else 
-			status = spacemit_i2c_read_reg(spacemit_i2c, REG_SR);
+		status = spacemit_i2c_read_reg(spacemit_i2c, REG_SR);
 		spacemit_i2c_clear_int_status(spacemit_i2c, status);
 		spacemit_i2c->i2c_status = status;
 
@@ -1183,17 +1172,13 @@ static int spacemit_i2c_pio_xfer(struct spacemit_i2c_dev *spacemit_i2c)
 
 		/* transmit empty */
 		if (likely(status & SR_ITE)) {
-			if ((spacemit_i2c->tx_cnt > 1) && (status & SR_MSD))
+			ret = spacemit_i2c_byte_xfer(spacemit_i2c);
+			if (unlikely(ret < 0))
 				break;
-			else {
-				ret = spacemit_i2c_byte_xfer(spacemit_i2c);
-				if (unlikely(ret < 0))
-					break;
-			}
 		}
-		
+
 		/* transaction done */
-		if (((status & SR_MSD) && !(status & SR_ITE)))
+		if (likely(status & SR_MSD))
 			break;
 
 		udelay(10);
-- 
2.47.0

